sssom/util.py

```diff
@@ -1170,55 +1170,58 @@ def get_prefixes_used_in_metadata(meta: MetadataType) -> List[str]:
 
 
 def filter_out_prefixes(
-    df: pd.DataFrame, filter_prefixes: List[str], features: list = KEY_FEATURES
+    df: pd.DataFrame,
+    filter_prefixes: List[str],
+    features: list = KEY_FEATURES,
+    require_all_prefixes: bool = False,
 ) -> pd.DataFrame:
-    """Filter any row where a CURIE in one of the key column uses one of the given prefixes.
+    """Filter out rows which contains a CURIE with a prefix in the filter_prefixes list.
 
-    :param df: Pandas DataFrame
+    :param df: Pandas DataFrame of SSSOM Mapping
     :param filter_prefixes: List of prefixes
     :param features: List of dataframe column names dataframe to consider
+    :param require_all_prefixes: If True, all prefixes must be present in a row to be filtered out
     :return: Pandas Dataframe
     """
     filter_prefix_set = set(filter_prefixes)
     rows = []
+    selection = all if require_all_prefixes else any
 
     for _, row in df.iterrows():
         prefixes = {get_prefix_from_curie(curie) for curie in row[features]}
-        # Confirm if none of the CURIEs in the list above appear in the filter_prefixes list.
-        # If TRUE, append row.
-        if not any(prefix in prefixes for prefix in filter_prefix_set):
+        if not selection(prefix in prefixes for prefix in filter_prefix_set):
             rows.append(row)
-    if rows:
-        return pd.DataFrame(rows)
-    else:
-        return pd.DataFrame(columns=features)
+
+    return pd.DataFrame(rows) if rows else pd.DataFrame(columns=features)
 
 
 def filter_prefixes(
-    df: pd.DataFrame, filter_prefixes: List[str], features: list = KEY_FEATURES
+    df: pd.DataFrame,
+    filter_prefixes: List[str],
+    features: list = KEY_FEATURES,
+    require_all_prefixes: bool = True,
 ) -> pd.DataFrame:
-    """Filter any row where a CURIE in one of the key column uses one of the given prefixes.
+    """Filter out rows which do NOT contain a CURIE with a prefix in the filter_prefixes list.
 
-    :param df: Pandas DataFrame
+    Args:
```

Comment:
`Args:` not needed.

Reply:
ah thanks, that was leftover from when i realised this wasn't Google style docstrings!!